dot/digest/digest.go

```diff
@@ -118,12 +122,14 @@ func (h *Handler) NextGrandpaAuthorityChange() (next uint) {
 func (h *Handler) HandleDigests(header *types.Header) {
 	for i, d := range header.Digest.Types {
 		val, ok := d.Value().(types.ConsensusDigest)
-		if ok {
-			err := h.handleConsensusDigest(&val, header)
-			if err != nil {
-				h.logger.Errorf("cannot handle digests for block number %d, index %d, digest %s: %s",
-					header.Number, i, d.Value(), err)
-			}
+		if !ok {
+			continue
+		}
+
+		err := h.handleConsensusDigest(&val, header)
+		if err != nil {
+			h.logger.Errorf("cannot handle digests for block number %d, index %d, digest %s: %s",
+				header.Number, i, d.Value(), err)
```

Comment:
Using the `val` and changed the `func (d *ConsensusDigest) String()` to `func (d ConsensusDigest) String()` so now the `%s` can call the `String()` method in both cases `val` or `d.Value()`, before it was not possible because the `String()` method belonging to the type `*ConsensusDigest` and `val` and `d.Value()` is of type `ConsensusDigest`.

Reply:
Ahhhh good job